cfgov/unprocessed/apps/cfpb-chart-builder/js/utils/get-data.js

```diff
@@ -9,30 +8,23 @@ const getData = (sources) => {
 
   const urls = sources.split(';');
 
-  const promises = urls.map(
-    (url) =>
-      new Promise((resolve, reject) => {
-        // Only prepend the data source base if it's a relative URL.
-        if (url.indexOf('http') !== 0 && url.indexOf('/') !== 0) {
-          url = DATA_SOURCE_BASE + url.replace('.csv', '.json');
-        }
-
-        if (cache.getItem(url)) {
-          /* Ensure UI isn't blocked when loading large shapefiles by making
-         cache resolver asynchronous https://stackoverflow.com/q/10180391 */
-          return setTimeout(() => resolve(cache.getItem(url)), 0);
-        }
-
-        return ajax({ url: url, type: 'json' }, function (resp) {
-          if (resp.error) {
-            reject(resp.error);
-            return;
-          }
-          cache.setItem(url, resp.data);
-          resolve(resp.data);
-        });
-      }),
-  );
+  const promises = urls.map((url) => {
+    // Only prepend the data source base if it's a relative URL.
+    if (url.indexOf('http') !== 0 && url.indexOf('/') !== 0) {
+      url = DATA_SOURCE_BASE + url.replace('.csv', '.json');
+    }
+
+    if (cache.getItem(url)) {
+      return Promise.resolve(cache.getItem(url));
+    }
+
+    return fetch(url).then((resp) => {
+      return resp.json().then((data) => {
+        cache.setItem(url, data);
+        return data;
+      });
+    });
+  });
 
   return Promise.all(promises);
```

Comment:
Do we need `Promise.all(…)` any longer, or can solely `promises` be returned? Not tested, but looking at the code it looks like we may have `Promise.all(…)` in there because the code had `new Promise(…)` and also the reference to xdr, which maybe created a promise of its own?

Reply:
getData allows `;` separated urls as source data and returns a single promise when they're all resolved. To preserve semantics, I think we should keep this as is.